### ISSUE TITLE
Docker-ize Setup

### ISSUE THREAD

**Issue by lxcodes** [coderie/coderie-site#15]
-- WAT?

![wait](https://cloud.githubusercontent.com/assets/62131/8606357/14fbc370-2659-11e5-8ef1-3a8dcdf154e3.gif)

No. Rly. Let's make that install easier? Discussion issue to judge support for changing from manual hope-and-pray to something maybe a bit more consistent?

**Comment by lxcodes** [comment]
I know Silva has been having some real serious issues, and I know how much of a pain Postgres is on mac. Just gauging interest in changing this and working documentation up around something more contained instead. Be it Docker or a Vagrant setup script.

/cc @machuga @matsilva @htmelvis

**Comment by machuga** [comment]
I have no idea how to docker properly but not against having a container or vm if it makes it easier for folk.  Lower ze barriers!  I'd recommend leveraging that site that lets you build Rails VM's easily and asking @redstonemercury nicely to review it to make sure it's not rubbish :D

**Comment by matsilva** [comment]
Docker could be interesting! I am pretty newish with ruby so I would have trouble trying to help dockerize it.

Or just create a task runner to automate the build.... any ruby task runners? bash script?

**Comment by lxcodes** [comment]
@matsilva Initial docker setup is already done actually =] -- More going to be issues around documentation and working with / tweaking it.

The problem with a task runner is actually getting everything to install and making sure the correct configs are in each file. Chef or Ansible might be able to handle a lot of the setup for us, but I'm not sure if people want to be running those on their machines.

VM would utilize Chef / Ansible as well but obviously wouldn't be destructive to their machine. Also a lot more tweak-able in the VM configuration, but I'm not sure everyone would be comfortable adding their tweaks to a build script.

**Comment by htmelvis** [comment]
I'm down to learn @al3x-edge . Whatever we do.

**Comment by lxcodes** [comment]
I've included basic documentation for Docker setup.  I would suggest using Kitematic on OS X for now.  With that in mind, add: `eval "$(docker-machine env dev)"` to your `~/.zshrc` or `~/.bashrc` in order to use the VM Kitematic uses for Docker.

**Comment by lxcodes** [comment]
I forgot a couple other steps, but back to work: `boot2docker ip` won't work as it doesn't work with Kitematic's vm instance. Boo. Anyway, Kitematic has gui elements to show you the IP of the container along with a little preview and logs.

![image](https://cloud.githubusercontent.com/assets/62131/8627220/7dab508e-2717-11e5-86f8-deb8a808f4d6.png)

By default currently, I have this running on port 3000.  If people have issues with other containers using port 3000, we can change it to something else.

![image](https://cloud.githubusercontent.com/assets/62131/8627230/9aa56cec-2717-11e5-9aff-b7e9eccada84.png)

**Comment by ohlincik** [comment]
When did web development became so complicated? We need to run a VM and a Docker container to run a Rails app? CRAZY!

**Comment by machuga** [comment]
Well something needs to hold the turbolinks together until we can wire the action cable through the coffeescript sass and around the haml, tying it off with the rspec active records.

**Comment by lxcodes** [comment]
lawls.  But rly. 

tl;dr programmers be masters of abstraction & layers.
![wave-abstraction](https://cloud.githubusercontent.com/assets/62131/8628752/8fa5e964-2723-11e5-8fc8-c4de703d11c6.gif)

--incomingwallotext--

Nothing on the install process of Ruby or Rails themselves and what can go wrong. But web development probably got this complicated once we started using dynamic languages on the server which would be PHP.  PHP when it first came out wasn't as user-friendly as it was today.  PHP just got so popular that eventually the modules for servers became integrated, well tested, and very simple to install because it was complicated initially.

Even C is more complicated than it used to be though.  Just natural tendency of what happens to code that is used longer and longer.

Sadly, the only reason we have a vm is because OS X hasn't implemented BSD functionality -.- and Windows will probably never get them.   Best part about Rails in Vagrant / Docker -- You can actually use it on Windows nicely instead of trying to development on it >.<

p.s. Spent more time looking for a gip than typing this.

**Comment by htmelvis** [comment]
you guys sound like an episode of dragon ball z. let's take things back to the roots. :) @al3x-edge I'm so glad you wrote that post and also that I saw that gif. Awesome. Thanks for taking the time to put that stuff together thats great. Have we nailed down the server yet because i'm down to donate if that isn't taken care of.

**Comment by machuga** [comment]
This seems appropriate.
![do it](https://lh3.googleusercontent.com/-MlZyK5-JqOg/VZLZFGif0oI/AAAAAAABJ1U/8WUbFWSSa-M/w506-h750/15%2B-%2B1)

**Comment by matsilva** [comment]
Nice work @al3x-edge , when I get a moment I will try to get this setup. Thanks for getting this put together!
@machuga thats awesome.

**Comment by lxcodes** [comment]
This has been completed. :smile_cat:
